### ISSUE TITLE
`ddev debug migrate-database mariadb:10.11` altered `config.yaml` more than needed

### ISSUE THREAD

**Issue by hanoii** [ddev/ddev#6847]
### Preliminary checklist

- [x] I am using the latest stable version of DDEV (see [upgrade guide](https://ddev.readthedocs.io/en/stable/users/install/ddev-upgrade/))
- [x] I have searched [existing issues](https://github.com/ddev/ddev/issues)
- [x] I have checked the [troubleshooting guide](https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/)
- [x] I have run `ddev debug test` to include output below

### Output of `ddev debug test`

I ran this command for the first time ever:

`ddev debug migrate-database mariadb:10.11`

All went fine but then I noticed the changed `config.yaml` had more than it should.

It basically brought in all changes by the add-ons present in the project, so the updated `config.yaml` was the merged result of all config.*.yaml present on this project.

### Expected Behavior

Only mariadb should change, along with maybe what `ddev config --auto` or `ddev config --update` does, but definitely not altering it based on other config.*.yaml files.

### Actual Behavior

Explained above.

**Comment by rfay** [comment]
Can you be more specific about what changed please? (Note that `ddev debug migrate-database` is in the `debug` section for a reason :)

**Comment by hanoii** [comment]
In this particular project, I only had https://github.com/hanoii/ddev-pimp-my-shell/blob/f8edb376716397863bd21dbfa2644895b2866b6d/config.pimp-my-shell.yaml, and all of that appeared on the main `config.yaml` file along the db change. `webimage_extra_packages` was even formatted as a single line array.

**Comment by rfay** [comment]
I am surprised. Thanks.

**Comment by rfay** [comment]
I can reproduce this with your exact case.

The reason is that we use GetActiveApp() in https://github.com/ddev/ddev/blob/66c3033fe0850211653efc73869f92cab805b61d/cmd/ddev/cmd/debug-migrate-database.go#L22

That's the full `app`, including `config.*.yaml`

Then we naively write the config back in https://github.com/ddev/ddev/blob/66c3033fe0850211653efc73869f92cab805b61d/cmd/ddev/cmd/debug-migrate-database.go#L66-L68

And what writes the whole thing.

I'm not sure what the exact fix is, but in this particular case we could do `app.ReadConfig("", false)` and change the database type and write. That, unfortunately, would miss the possible case where the database type and version were specified in `config.*.yaml`

**Comment by GuySartorelli** [comment]
Is there a mechanism to identify which config file contains a given setting?

**Comment by rfay** [comment]
No, I don't think so. app.ReadConfig has the `includeOverrides` arg:

https://github.com/ddev/ddev/blob/7a0c06cd1d9cc923e07e88074a6b3010b5087ada/pkg/ddevapp/config.go#L314-L338

It would be a bit of a violation for DDEV to touch a `config.*.yaml` though, as those aren't ever changed by DDEV directly.

**Comment by hanoii** [comment]
> I'm not sure what the exact fix is, but in this particular case we could do app.ReadConfig("", false) and change the database type and write.

I think this is OK, isn't this the same that `ddev config --update` does?

> That, unfortunately, would miss the possible case where the database type and version were specified in config.*.yaml

I think that this is also currently the case: assume a mariadb of 10.4 on config.yaml, and a mariadb of 10.6 on config.db.yaml, running `ddev debug migrate-database mariadb:10.11` would write 10.11 to config.yaml, but  config.db.yaml would still be 10.6 which would continue to overwrite it.

> It would be a bit of a violation for DDEV to touch a config.*.yaml though, as those aren't ever changed by DDEV directly.

Agreed, and the same reasoning should apply the other way around, it shouldn't care of what config.*.yaml does. So, if a config.something.yaml set a specific db (which I think should be very very rare), then there should be a reason for doing so even though the migration will be attempted, the config.something.yaml will continue to leave the db as it expects it to. At that case is up for the user to decide what to do.

Anything further I think is overcomplicating things. We could, if it's possible, process all config.*.yaml files independently in the same way we are doing now (without the main config.yaml( and then check to see if a db is being set, and warn accordingly.

But I think it's too much for a debug command.

**Comment by rfay** [comment]
It would probably be legit for `ddev debug migrate-database` to app.ReadConfig(false), then change the type/version, then do its job.  That seems like a modest one-line change here that is in an off-the-beaten-track place. But one has to wonder where else this conflict might exist (read with includeOverrides, then write config). It doesn't seem like with current informal policies this is an easily solvable problem. But I think it would be fine to solve it here.

**Comment by GuySartorelli** [comment]
Perhaps if the database isn't in the main config file there could be a message output to the terminal that no change was made.
